include/wx/generic/private/grid.h

```diff
@@ -1280,6 +1294,235 @@ TryGetValueAsDate(wxDateTime& result,
 
 #endif // wxUSE_DATETIME
 
+//=============================================================================
+// PolyPolygon & PolyPolygonHelper classes
+//=============================================================================
+
+// A simple interface used by wxGrid::DrawSelectionOverlay() as a helper to draw
+// the grid selection overlay.
+class PolyPolygon
+{
+public:
+    PolyPolygon() = default;
+    ~PolyPolygon() = default;
+
+    // Return the number of polygons to draw.
+    size_t GetSize() const { return m_counts.size(); }
+
+    const int* GetCounts() const { return m_counts.data(); }
+
+    const wxPoint* GetPoints() const { return m_points.data(); }
+
+    void Append(const std::vector<wxPoint>& points);
+
+    // Return the bounding box of the visible selected blocks. Return empty
+    // rectangle if there is no selection.
+    wxRect GetBoundingBox() const;
+
+    void SetBoundingBox(const wxRect& rect);
+
+private:
+    void CalcBoundingBox(const std::vector<wxPoint>& points);
+
+    // m_counts contains the number of points in each of the polygons in m_points.
```

Comment:
So the data structure here is a vector of points in all polygons stored as a single continuous vector? I.e. if we have 2 polygons P1 and P2, `m_points` contains P1_first, P1_second, ..., P1_last, P2_first, ..., P2_last?

Is this done for efficiency reasons? Because for me a natural approach would be to just have `vector<vector<wxPoint>> m_polygons` and I wonder why is it not being used here?

Reply:
> So the data structure here is a vector of points in all polygons stored as a single continuous vector? I.e. if we have 2 polygons P1 and P2, `m_points` contains P1_first, P1_second, ..., P1_last, P2_first, ..., P2_last?
> 

Yes.

> Is this done for efficiency reasons? Because for me a natural approach would be to just have `vector<vector<wxPoint>> m_polygons` and I wonder why is it not being used here?

Well, the data structure is modeled in such a way that we can use it to draw a poly-polygon using [DrawPolyPolygon()](https://docs.wxwidgets.org/3.2/classwx_d_c.html#ae760da96008debe42e18195d8ac8df3d) in a simple and straightforward way. I don't see how a `vector<vector<wxPoint>>` would simplify things here to be honest?

Reply:
Sorry, I forgot about this `DrawPolyPolygon()` convention, I thought the polygons would be drawn one by one, but if we use this function then storing it like this makes sense, thanks, even if using a vector of vectors would still definitely be simpler (because a polygon is a vector of points, and so many polygons is a vector of such vectors).